examples/nftvm/storage/nft_storage.go

```diff
@@ -0,0 +1,273 @@
+	if err != nil {
+		return 0, err
+	}
+
+	// Write instance to state
+	if err = SetNFTInstance(ctx, mu, collectionAddress, numOfInstances, owner, metadata, false); err != nil {
+		return 0, err
+	}
+
+	// Update parent collection in state
+	if err = SetNFTCollection(ctx, mu, collectionAddress, name, symbol, collectionMetadata, numOfInstances+1, collectionOwner); err != nil {
+		return 0, nil
+	}
+
+	return numOfInstances, nil
+}
+
+// Writes NFT instance to state
+func SetNFTInstance(
+	ctx context.Context,
+	mu state.Mutable,
+	collectionAddress codec.Address,
+	instanceNum uint32,
+	owner codec.Address,
+	metadata []byte,
+	isListedOnMarketplace bool,
+) error {
+	instanceStateKey := InstanceStateKey(collectionAddress, instanceNum)
+
+	metadataLen := len(metadata)
+	v := make([]byte, codec.AddressLen+consts.Uint16Len+metadataLen+consts.Uint16Len)
+
+	// Insert instance owner
+	copy(v, owner[:])
+
+	// Insert metadata
+	binary.BigEndian.PutUint16(v[codec.AddressLen:], uint16(metadataLen))
+	copy(v[codec.AddressLen+consts.Uint16Len:], metadata)
+
+	// Insert marketplace indicator
+	if isListedOnMarketplace {
+		binary.BigEndian.PutUint16(v[codec.AddressLen+consts.Uint16Len+metadataLen:], 1)
+	} else {
+		binary.BigEndian.PutUint16(v[codec.AddressLen+consts.Uint16Len+metadataLen:], 0)
+	}
+
+	return mu.Insert(ctx, instanceStateKey, v)
+}
+
+func GetNFTInstance(
+	ctx context.Context,
+	f ReadState,
+	collectionAddress codec.Address,
+	instanceNum uint32,
+) (owner codec.Address, metadata []byte, isListedOnMarketplace bool, err error) {
+	k := InstanceStateKey(collectionAddress, instanceNum)
+	values, errs := f(ctx, [][]byte{k})
+	return innerGetNFTInstance(values[0], errs[0])
+}
+
+func GetNFTInstanceNoController(
+	ctx context.Context,
+	mu state.Mutable,
+	collectionAddress codec.Address,
+	instanceNum uint32,
+) (owner codec.Address, metadata []byte, isListedOnMarketplace bool, err error) {
+	value, err := mu.GetValue(ctx, InstanceStateKey(collectionAddress, instanceNum))
+	if err != nil {
+		return codec.EmptyAddress, []byte{}, false, err
+	}
+	return innerGetNFTInstance(value, err)
+}
+
+func innerGetNFTInstance(
+	v []byte,
+	err error,
+) (owner codec.Address, metadata []byte, isListedOnMarketplace bool, e error) {
+	if err != nil {
+		return codec.EmptyAddress, []byte{}, false, err
+	}
+	// Extract owner
+	owner = codec.Address(v[:codec.AddressLen])
+	// Extract metadata
+	metadataLen := binary.BigEndian.Uint16(v[codec.AddressLen:])
+	metadata = v[codec.AddressLen+consts.Uint16Len : codec.AddressLen+consts.Uint16Len+metadataLen]
+	// Extract marketplace indicator
+	marketplaceIndicator := binary.BigEndian.Uint16(v[codec.AddressLen+consts.Uint16Len+metadataLen:])
+	if marketplaceIndicator == uint16(1) {
+		isListedOnMarketplace = true
+	} else if marketplaceIndicator == uint16(0) {
+		isListedOnMarketplace = false
+	} else {
+		return codec.EmptyAddress, []byte{}, false, ErrCorruptInstanceMarketplaceIndicator
+	}
```

Comment:
can we replace this with a switch statement and replace `uint16(1)` and `uint16(0)` with constants? I'd recommend using `iota` and following the style here: https://github.com/ava-labs/avalanchego/blob/f4d8a3c0647b2b96aa0dbf5de6961d609d030b88/snow/state.go#L13